### ISSUE TITLE
sparse matrix methods

### ISSUE THREAD

**Issue by topepo** [tidymodels/hardhat#100]
It would be nice to have`dgCMatrix` support (now that I have an application that can really use that)

**Comment by DavisVaughan** [comment]
I would need a little bit more information about this. Currently `mold()` forces any matrix input into a data frame. That makes the output from `mold()` consistent across all interfaces (i.e. the `predictors` and `outcomes` are always data frames).

It would be pretty inconsistent to return a `dgCMatrix` in the `predictors` slot even if had performance improvements

**Comment by juliasilge** [comment]
I looked at this a bit yesterday and realized that it is going to be a significant undertaking. Being able get to something like this:

```r
new_default_recipe_blueprint(
    mold = mold,
    forge = forge,
    intercept = intercept,
    allow_novel_levels = allow_novel_levels,
    composition = composition,
    fresh = fresh
  )
```
is going to be a significant improvement in quality of life in several domains, though.

**Comment by DavisVaughan** [comment]
Closed by #150 (closed enough for now, at least). We can revisit in the future as needed
